pkg/collector/data/examples_generator.go

```diff
@@ -276,25 +275,90 @@ type Rule struct {
 	Dest                 string
 	DestinationsExcluded bool
 	Services             []string
+	Conn                 *netset.TransportSet
 	Action               string
 	Direction            string // if not set, used as default with "IN_OUT"
 	Description          string
 }
 
-func (r *Rule) toNSXRule() *nsx.Rule {
-	return &nsx.Rule{
-		DisplayName:          &r.Name,
-		RuleId:               &r.ID,
-		Action:               (*nsx.RuleAction)(&r.Action),
-		SourceGroups:         []string{r.Source},
-		DestinationGroups:    []string{r.Dest},
-		SourcesExcluded:      r.SourcesExcluded,
-		DestinationsExcluded: r.DestinationsExcluded,
-		Services:             r.Services,
-		Direction:            r.directionStr(),
-		Scope:                []string{AnyStr}, // TODO: add scope as configurable
-		Description:          &r.Description,
+func (r *Rule) toCollectorRule() collector.Rule {
+	services, entries := serviceEntries(r.Services, r.Conn)
+	return collector.Rule{
+		Rule: nsx.Rule{
+			DisplayName:          &r.Name,
+			RuleId:               &r.ID,
+			Action:               (*nsx.RuleAction)(&r.Action),
+			SourceGroups:         []string{r.Source},
+			DestinationGroups:    []string{r.Dest},
+			SourcesExcluded:      r.SourcesExcluded,
+			DestinationsExcluded: r.DestinationsExcluded,
+			Services:             services,
+			Direction:            r.directionStr(),
+			Scope:                []string{AnyStr}, // TODO: add scope as configurable
+			Description:          &r.Description,
+		},
+		ServiceEntries: entries,
+	}
+}
+
+var codeToProtocol = map[int]nsx.L4PortSetServiceEntryL4Protocol{
+	netset.UDPCode: nsx.L4PortSetServiceEntryL4ProtocolUDP,
+	netset.TCPCode: nsx.L4PortSetServiceEntryL4ProtocolTCP,
+}
+
+func serviceEntries(services []string, conn *netset.TransportSet) ([]string, collector.ServiceEntries) {
```

Comment:
can add some documentation about this function?

Reply:
done